components/tidb_query/src/rpn_expr/impl_string.rs

```diff
@@ -197,6 +198,24 @@ pub fn right(lhs: &Option<Bytes>, rhs: &Option<Int>) -> Result<Option<Bytes>> {
     }
 }
 
+#[rpn_fn(capture = [extra])]
+#[inline]
+pub fn upper(extra: &RpnFnCallExtra, arg: &Option<Bytes>) -> Result<Option<Bytes>> {
```

Comment:
Hi, for this case, prefer to change TiDB first, separate `upper_binary` and `upper` into two signatures. You will be rewarded 150 points for doing these tasks altogether (not necessary in one PR):

1. change TiDB to introduce separate signature for binary.
2. change TiKV non-vectorized implementation to support binary upper.
3. vectorize both upper and upper_binary.

@lonng PTAL